core/tests/protractor/learnerFlow.js

```diff
@@ -0,0 +1,460 @@
+        collectionEditorPage.setCategory('Algebra');
+        collectionEditorPage.saveChanges();
+        users.logout();
+      }
+      var PLAYER_USERNAME = 'collectionPlayerDesktopAndMobile';
+      users.createAndLoginUser(
+        'collectionPlayerDesktopAndMobile@learnerFlow.com', PLAYER_USERNAME);
+      if (browser.isMobile) {
+        browser.get('/collection/1');
+        waitFor.pageToFullyLoad();
+        expect(browser.getCurrentUrl()).toEqual(
+          'http://localhost:9001/collection/1');
+      } else {
+        libraryPage.get();
+        libraryPage.findCollection('Collection Player Test');
+        libraryPage.playCollection('Collection Player Test');
+      }
+    });
+
+  it('displays incomplete and completed explorations', function() {
+    // Create explorations for desktop testing.
+    if (!browser.isMobile) {
+      users.createAndLoginUser('originalCreator@learnerDashboard.com',
+        'originalCreator');
+      // Create exploration 'Dummy Exploration'
+      createDummyExplorationOnDesktop();
+      // Create a second exploration named 'Test Exploration'.
+      workflow.createAndPublishExploration(
+        'Test Exploration',
+        'Astronomy',
+        'To expand the horizon of the minds!',
+        'English'
+      );
+      users.logout();
+    }
+    users.createAndLoginUser('learner@learnerDashboard.com',
+      'learnerlearnerDashboard');
+    // Go to 'Dummy Exploration'.
+    // Leave this exploration incomplete.
+    if (browser.isMobile) {
+      browser.get('/explore/24');
+      waitFor.pageToFullyLoad();
+      clickContinueButton();
+    } else {
+      libraryPage.get();
+      libraryPage.findExploration('Dummy Exploration');
+      libraryPage.playExploration('Dummy Exploration');
+      explorationPlayerPage.expectExplorationNameToBe('Dummy Exploration');
+      explorationPlayerPage.submitAnswer('Continue', null);
+      explorationPlayerPage.expectExplorationToNotBeOver();
+    }
+    // User clicks on Oppia logo to leave exploration.
+    oppiaLogo.click();
+    general.acceptAlert();
+    // Wait for /learner_dashboard to load.
+    waitFor.pageToFullyLoad();
+
+    // Go to 'Test Exploration'.
+    // Complete this exploration.
+    if (browser.isMobile) {
+      browser.get('/explore/22');
+      waitFor.pageToFullyLoad();
+    } else {
+      libraryPage.get();
+      libraryPage.findExploration('Test Exploration');
+      libraryPage.playExploration('Test Exploration');
+      explorationPlayerPage.expectExplorationNameToBe('Test Exploration');
+    }
+    oppiaLogo.click();
+    waitFor.pageToFullyLoad();
+    // Learner Dashboard should display 'Dummy Exploration'
+    // as incomplete.
+    checkIncompleteExplorationSection('Dummy Exploration');
+
+    // Learner Dashboard should display 'Test Exploration'
+    // exploration as complete.
+    checkCompleteExplorationSection('Test Exploration');
+
+    // Now complete the 'Dummmy Exploration'.
+    if (browser.isMobile) {
```

Comment:
Ditto. I mean, you could change the exploration titles as needed... right? So that they look the same from the point of the library page?

Reply:
Yes, done.